lib/parsec/empysupport.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2017 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""cylc support for the EmPy template processor
+
+Importing code should catch ImportError in case EmPy is not installed.
+"""
+
+from StringIO import StringIO
+import em
+import os
+import sys
+
+
+class EmPyError(Exception):
+    """Wrapper class for EmPy exceptions."""
+
+    def __init__(self, exc, lineno):
+        Exception.__init__(self, exc)
+        self.lineno = lineno
+
+
+def empyprocess(flines, dir_, template_vars=None):
+    """Pass configure file through EmPy processor."""
+
+    cwd = os.getcwd()
+
+    try:
+        os.chdir(dir_)
+        ftempl = StringIO('\n'.join(flines))
+        xtempl = StringIO()
+        interpreter = em.Interpreter(output=em.UncloseableFile(xtempl))
+        interpreter.file(ftempl, '<template>', template_vars)
+    except Exception as exc:
```

Comment:
Do lines 43-45 need to be inside this try block?  If they raise an exception, it is presumably not due to an EmPy error.